sentry/src/lib.rs

```diff
@@ -510,19 +512,38 @@ pub mod test_util {
         Application,
     };
 
+    /// This guard holds the Redis and Postgres pools taken from their respective Pool of pools.
+    ///
+    /// This ensures that they will not be dropped which will cause tests to fail randomly.
+    pub struct ApplicationGuard {
+        pub app: Application<Dummy>,
+        #[allow(dead_code)]
+        redis_pool: deadpool::managed::Object<crate::db::redis_pool::Manager>,
+        #[allow(dead_code)]
+        db_pool: deadpool::managed::Object<crate::db::tests_postgres::Manager>,
```

Comment:
Because the `Object` was getting dropped tests failed randomly.
Object hold the Pool of Pools, so we must hold these values until the Application struct get drop itself